src/libpriv/rpmostree-kernel.c

```diff
@@ -381,11 +375,8 @@ rpmostree_run_dracut (int     rootfs_dfd,
     (void) unlinkat (rootfs_dfd, rebuild_from_initramfs, 0);
 
   ret = TRUE;
-  *out_initramfs_tmpfd = tmp_fd; tmp_fd = -1;
-  *out_initramfs_tmppath = g_steal_pointer (&tmpfile_path);
+  *out_initramfs_tmpf = tmpf; tmpf.initialized = FALSE; /* Transfer */
```

Comment:
Hmm, I feel like this warrants a `glnx_steal_tmpf()` helper to abstract over these details.